pennylane/tape/qscript.py

```diff
@@ -935,11 +935,15 @@ def shape(self, device):
         if not qml.active_return():
             return self._shape_legacy(device)
 
-        shots = (
-            Shots(device._raw_shot_sequence)
-            if device.shot_vector is not None
-            else Shots(device.shots)
-        )
+        if isinstance(device, qml.devices.experimental.Device):
+            shots = self.shots
+            device = self  # most MPs use `device.wires`, one checks `device.cutoff`
```

Comment:
This code has a very ugly smell. Do we expect this to only be a temporary change, or is more long-term?

Reply:
tbh I am not a fan of this whole tape.shape -> MP.shape thing so I wanted to change as little as possible, and this was the way to do it. if you have another proposal i'm happy to take it, but I don't think it's worth spending toooo much time on. the only other path I saw was refactoring everything, and Mudit already tried it once in the past.

perhaps if we get rid of support from the old device, this will go away. or if we make a dedicated task out of it. but otherwise, I think this will remain for a while.

Reply:
Sure, I get that it's outside the scope of this PR. Can we have a more detailed comment here saying that `wires` used to be on the device but now it's on the tape, hence why this is necessary?